lib/library.js

```diff
@@ -31,18 +31,20 @@ var EXT = Library.EXT = {
  * ForeignFunction.
  */
 
-function Library (libfile, funcs, lib) {
-  debug('creating Library object for', libfile)
+function Library (dl, funcs, lib) {
+  debug('creating Library object for', dl)
 
-  if (libfile && libfile.indexOf(EXT) === -1) {
-    debug('appending library extension to library name', EXT)
-    libfile += EXT
+  if (!(dl instanceof DynamicLibrary)) {
```

Comment:
Let's do `if ('string' === typeof dl)` instead. I'm paranoid about `instanceof` and try to avoid it when possible.